### DISCUSSION TITLE
Exclude endpoint from the others tests

### DISCUSSION THREAD

**Discussion by abrevet-dev** [schemathesis/schemathesis#1457] [Q&A]
Hi,

I have a use case where I want to customize the test of only one endpoint:

```python
# specific endpoint
@schema.parametrize(endpoint="/api/users")  
def test_users_api(case):
    case.headers = case.headers or {}
    case.headers["custom"] = "custom"
    ...

# rest of the API
@schema.parametrize()  
def test_api(case):
    ...
```
But with this code, the endpoint `/api/users` is also tested in the function `test_api` and fail due to missing headers.

How to exclude the endpoint `/api/users` from the second test `test_api` ?

I appreciate your help

**Comment by Stranger6667** [comment]
I think it could be solved with the [built-in overriding](https://schemathesis.readthedocs.io/en/stable/python.html#overriding-test-data) in some cases (and it might be just the right fit for what you've described).

Alternatively, the endpoint could be excluded with a negative lookahead in a regex. This is still hacky and I want to provide a more convenient API to achieve this goal, but it is still WIP.
Such a regex could look like this - `^(?!\/api\/users$).*`
